src/levels/sparsetrianglelevels.jl

```diff
@@ -66,17 +56,24 @@ function Base.show(io::IO, lvl::SparseTriangleLevel{N, Ti}) where {N, Ti}
 end 
 
 function display_fiber(io::IO, mime::MIME"text/plain", fbr::SubFiber{<:SparseTriangleLevel{N}}, depth) where {N}
-    qos = (fbr.lvl.shape * (fbr.lvl.shape + 1)) >>> 1
+    qos = simplex(fbr.lvl.shape, N)
     crds = 1:qos
 
-    function print_coord(io, q)
-        j = ceil(sqrt(2 * q + 0.25) - 0.5)
-        i = q - (j-1) * j / 2
-        print(io, i, ", ", j)
-    end 
+    #when n = 3, this is the qth element of [(i, j, k) for k = 1:3 for j = 1:k for i = 1:j]
+    function get_coord(q, n, k)
```

Comment:
display_fiber is admittedly a bit tricky to implement correctly, I've made the necessary N-D generalization here, avoiding sqrt for a search (which is not yet a binary search but could be in the future)